aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1354,10 +1353,6 @@ Tensor nanmean(
     at::OptionalIntArrayRef dim,
     bool keepdim,
     optional<ScalarType> opt_dtype) {
-  TORCH_CHECK(
```

Comment:
@Skylion007 is right. This check ensures that `nanmean` only works with `floating` inputs.

Example @khushi-411 presented used `nansum` which works with integral inputs.

```python
import torch

t = torch.tensor([1,2 ,3])

torch.nanmean(t)  # RuntimeError: nanmean(): expected input to have floating point dtype but got Long

```

Reply:
> @Skylion007 is right. This check ensures that nanmean only works with floating inputs. (NOTE: mean also doesn't support integral dtypes)

Yes, I'm sorry.

I've updated it now.